### PR TITLE
refactor: cleanup Zhegalkin

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala [hunk 1]
```diff
@@ -51,7 +52,7 @@ object ZhegalkinExpr {
       }
 
       // General case:
-      ZhegalkinExpr(cst, ts)
+      ZhegalkinExpr(cst, ts.sortBy(t => t.vars.size))
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala:55]: Behavioral change :O 

Should this have a comment about why we sort?

File: main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala [hunk 2]
```diff
@@ -145,10 +146,15 @@ object ZhegalkinExpr {
...
-  // TODO: Docs
-  private def computeUnion(a: ZhegalkinExpr, b: ZhegalkinExpr): ZhegalkinExpr = {
-    /** a ∪ b = a ⊕ b ⊕ (a ∩ b) */
-    mkXor(mkXor(a, b), mkInter(a, b))
+  /**
+    * Computes the union of the given two Zhegalkin expressions `e1` and `e2`.
+    */
+  private def computeUnion(e1: ZhegalkinExpr, e2: ZhegalkinExpr): ZhegalkinExpr = {
+    // a ∪ b = 1 ⊕ (1 ⊕ a)(1 ⊕ b)
+    //mkXor(ZhegalkinExpr.one, mkInter(mkXor(ZhegalkinExpr.one, e1), mkXor(ZhegalkinExpr.one, e2)))
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala:154]: This seems less clean than before.

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala:154]: Its still a work in progress :)

File: main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala:55]: Its just for performance. Ill comment it in the next round.

**Pull request merged**